src/frequenz/client/common/microgrid/__init__.py

```diff
@@ -1,4 +1,51 @@
 # License: MIT
-# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+# Copyright © 2024 Frequenz Energy-as-a-Service GmbH
 
-"""Frequenz microgrid definition."""
+"""Module to define the microgrid used with the common client."""
+
+from __future__ import annotations  # required for constructor type hinting
+
+from dataclasses import dataclass
+from typing import List, Self
```

Comment:
Just use the built-in `list` type for type hints, `List` is deprecated.

```suggestion
from typing import Self
```